mtools/mloginfo/sections/connection_section.py

```diff
@@ -51,6 +71,18 @@ def run(self):
                     ip, _ = tokens[3].split(':')
                 ip_opened[ip] += 1
 
+                if genstats:
+                    dt = logevent.datetime
+
+                    #initialize using default constructor
+                    ipwise_min_connection_duration[ip]
+                    ipwise_max_connection_duration[ip]
+
+                    connid = tokens[4].strip('#')
+
+                    if dt != None:                  
+                        connections_start[connid] = dt           
```

Comment:
Should this be checking if there is already a start datetime for the given connection ID?

If there are multiple start datetimes found for the same connection ID should that raise an exception?

Reply:
Done will raise and exception now.

Reply:
will also raise exception if duplicate end connection ids

Reply:
I'm trying to understand why we need to raise an exception for this? I just encountered this exception on some 3.4.23 logs. Couldn't this just be a datetime precision issue?

Reply:
Ah, can this be raised when the same connection ID is reused during the same log period? The logs I have are for 24 hours but has a very high connection churn, so it's not impossible that this will happen on perfectly OK log files.

Reply:
Btw, filtering to a 1 hour period worked around the exception. Maybe the exception message should state the connection number that repeats?